modules/gapi/perf/common/gapi_video_perf_tests_inl.hpp

```diff
@@ -178,47 +178,53 @@ PERF_TEST_P_(BackgroundSubtractorPerfTest, TestPerformance)
         thr = 16.;
     else if (opType == gvideo::TYPE_BS_KNN)
         thr = 400.;
-    const gvideo::BackgroundSubtractorParams bsp(opType, histLength, thr, detectShadows, learningRate);
+    const gvideo::BackgroundSubtractorParams bsp(opType, histLength, thr, detectShadows,
+                                                 learningRate);
 
-    // Video source declaration
-    cv::VideoCapture cap;
-    if (!cap.open(findDataFile(filePath))) throw SkipTestException("Video file can not be opened");
-    cv::Mat frame;
-    cap >> frame;
+    // Retrieving frames
+    std::vector<cv::Mat> frames;
+    frames.reserve(testNumFrames);
+    {
+        cv::Mat frame;
+        cv::VideoCapture cap;
+        if (!cap.open(findDataFile(filePath)))
+            throw SkipTestException("Video file can not be opened");
+        for (std::size_t i = 0; i < testNumFrames && cap.read(frame); i++)
```

Comment:
testNumFrames - is it "warm up" frames in this case?

Reply:
No, I've changed the structure of test - no more "warm ups". This amount is the amount of frames for which as a whole we measure the performance.